ground/src/main/java/com/google/android/ground/persistence/remote/firebase/schema/SurveysCollectionReference.kt

```diff
@@ -31,9 +33,8 @@ class SurveysCollectionReference internal constructor(ref: CollectionReference)
 
   fun survey(id: String) = SurveyDocumentReference(reference().document(id))
 
-  suspend fun getReadable(user: User): List<Survey> =
-    runQuery(reference().whereIn(FieldPath.of(ACL_FIELD, user.email), Role.valueStrings())) {
-      doc: DocumentSnapshot ->
-      SurveyConverter.toSurvey(doc)
+  fun getReadable(user: User): Flow<List<Survey>> =
+    reference().whereIn(FieldPath.of(ACL_FIELD, user.email), Role.valueStrings()).snapshots().map {
+      it.documents.map { doc -> SurveyConverter.toSurvey(doc) }
```

Comment:
Great!!! This also solves the problem of manually canceling listening to updates.